app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -402,9 +401,19 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
                             components.core.store.state.selectedTab?.content?.private
                                 ?: components.settings.openLinksInAPrivateTab
 
-                        val session = Session(url, shouldCreatePrivateSession)
-                        components.core.sessionManager.add(session, true, engineSession)
-                        session.id
+                        if (shouldCreatePrivateSession) {
+                            components.useCases.tabsUseCases.addPrivateTab(
+                                url = url,
+                                selectTab = true,
+                                engineSession = engineSession
+                            )
+                        } else {
+                            components.useCases.tabsUseCases.addTab(
+                                url = url,
+                                selectTab = true,
+                                engineSession = engineSession
+                            )
+                        }
```

Comment:
Maybe we could have a use case for this on AC

Reply:
Thought about that too but it really would only save the if statement. Might still be worth it but since we want to remove those overrides here anyway as soon as the refactor is complete let's land and revisit.

Reply:
👍🏽